python/langsmith/integrations/otel/processor.py

```diff
@@ -0,0 +1,219 @@
+        Args:
+            url: OTLP endpoint URL. Defaults to {LANGSMITH_ENDPOINT}/otel/v1/traces.
+            api_key: LangSmith API key. Defaults to LANGSMITH_API_KEY env var.
+            parent: Parent identifier (e.g., "project_name:test").
+                Defaults to LANGSMITH_PARENT env var.
+            headers: Additional headers to include in requests.
+            **kwargs: Additional arguments passed to OTLPSpanExporter.
+        """
+        base_url = ls_utils.get_api_url(None)
+        # Ensure base_url ends with / for proper joining
+        if not base_url.endswith("/"):
+            base_url += "/"
+        endpoint = url or urljoin(base_url, "otel/v1/traces")
+        api_key = api_key or ls_utils.get_api_key(None)
+        project = project or ls_utils.get_tracer_project()
+        headers = headers or {}
+
+        if not api_key:
+            raise ValueError(
+                "API key is required. Provide it via api_key parameter or "
+                "LANGSMITH_API_KEY environment variable."
+            )
+
+        if not project:
+            project = "default"
+            logging.info(
+                "No project specified, using default. "
+                "Configure with LANGSMITH_PROJECT environment variable or "
+                "project parameter."
+            )
+
+        exporter_headers = {
+            "x-api-key": api_key,
+            **headers,
+        }
+
+        if project:
+            exporter_headers["Langsmith-Project"] = project
+
+        self.project = project
+
+        super().__init__(endpoint=endpoint, headers=exporter_headers, **kwargs)
+
+
+class OtelSpanProcessor:
+    """A span processor for adding LangSmith to OpenTelemetry setups.
+
+    This class combines the OtelExporter and BatchSpanProcessor
+    into a single processor that can be added to any TracerProvider.
+
+    Use this when:
+    1. You already have OpenTelemetry initialized with other tools
+    2. You want to add LangSmith alongside existing OTEL exporters
+
+    Examples:
+        # Fresh OpenTelemetry setup (LangSmith only):
+        from langsmith.integrations.otel import configure
+        configure(api_key="your-key", project="your-project")
+
+        # Add LangSmith to existing OpenTelemetry setup:
+        from opentelemetry import trace
+        from langsmith.integrations.otel.processor import OtelSpanProcessor
+
+        # Get your existing TracerProvider (already set by other tools)
+        provider = trace.get_tracer_provider()
+
+        # Add LangSmith processor alongside existing processors
+        langsmith_processor = OtelSpanProcessor(
+            project="your-project",
+        )
+        provider.add_span_processor(langsmith_processor)
+    """
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        project: Optional[str] = None,
+        url: Optional[str] = None,
+        headers: Optional[dict[str, str]] = None,
+        SpanProcessor: Optional[type] = None,
```

Comment:
OOC why would someone want something other than `BatchSpanProcessor`? I guess if you had more than one exporter?